auth.go

```diff
@@ -106,7 +122,27 @@ func New(opts Options) (Service, error) {
 		keyFile:      opts.KeyFile,
 		auths:        make(chan *slack.OAuthResponse, 1),
 		api:          &slackAPIWrapper{},
-	}, nil
+	}
+
+	return slackAuthService.configureButton(opts.ButtonTpl, opts.Scopes)
+}
+
+func (s *slackAuth) configureButton(buttonTpl string, scopes []string) (*slackAuth, error) {
+	if len(buttonTpl) > 0 {
+		buttonTpl, err := readTemplate(buttonTpl)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(scopes) == 0 {
+			return nil, errors.New("At least one scope needed")
+		}
+
+		s.scopes = strings.Join(scopes, ",")
+		s.buttonTpl = buttonTpl
+	}
+
+	return s, nil
```

Comment:
Fixed.
I was trying to make the method chainable, but since it returns an error, the method chaining is not possible...